esmvaltool/utils/cmorizers/obs/cmorize_obs_WOA.py

```diff
@@ -1,122 +1,100 @@
-    if var == 'o2':
-        cube.data = cube.data * 44.661 / 1000.  # Convert from ml/l to mol/m^3
+    logger.info("Fixing data ...")
+    with constant_metadata(cube):
+        mll_to_mol = ['po4', 'si', 'no3']
+        if var in mll_to_mol:
+            cube /= 1000.  # Convert from ml/l to mol/m^3
+        elif var == 'thetao':
+            cube += 273.15  # Convert to Kelvin
+        elif var == 'o2':
+            cube *= 44.661 / 1000.  # Convert from ml/l to mol/m^3
     return cube
 
 
-def extract_variable(var, raw_file, out_dir, yr):
+def extract_variable(var_info, raw_info, out_dir, attrs, year):
     """Extract to all vars."""
-    cubes = iris.load(raw_file)
-    field = FIELDS[var]
+    var = var_info.short_name
+    cubes = iris.load(raw_info['file'])
+    rawvar = raw_info['name']
+
     for cube in cubes:
-        if cube.long_name == field:
-            cube.standard_name = STANDARD_NAMES[var]
-            cube.long_name = LONG_NAMES[var]
-            cube.var_name = var
-            _convert_timeunits(cube, yr)
+        if cube.var_name == rawvar:
+            _fix_var_metadata(cube, var_info)
+            _convert_timeunits(cube, year)
             _fix_coords(cube)
-            _roll_cube_data(cube, 180, -1)
             _fix_data(cube, var)
-            _fix_metadata(cube, var)
-            _add_metadata(cube, proj)
-            _save_variable(cube, var, out_dir, yr, proj)
+            _set_global_atts(cube, attrs)
+            _save_variable(
+                cube, var, out_dir, attrs, unlimited_dimensions=['time'])
 
 
 def cmorization(in_dir, out_dir):
     """Cmorization func call."""
-    logger.info("Starting cmorization for WOA OBS files: Tier2")
+    cmor_table = CFG['cmor_table']
+    glob_attrs = CFG['attributes']
+
+    logger.info("Starting cmorization for Tier%s OBS files: %s",
+                glob_attrs['tier'], glob_attrs['dataset_id'])
     logger.info("Input data from: %s", in_dir)
     logger.info("Output will be written to: %s", out_dir)
 
     # run the cmorization
-    for var in ALL_VARS:
-        if not os.path.exists(out_dir):
-            os.path.makedirs(out_dir)
-        for yr in ALL_YEARS:
+    for var, vals in CFG['variables'].items():
+        yr = None
+        for yr in CFG['custom']['years']:
             file_suffix = str(yr)[-2:] + '_' + str(yr + 1)[-2:] + '.nc'
-            raw_file = os.path.join(in_dir, VAR_TO_FILENAME[var] + file_suffix)
-            logger.info("CMORizing var %s in file %s", var, raw_file)
-            extract_variable(var, raw_file, out_dir, yr)
+            inpfile = os.path.join(in_dir, vals['file'] + file_suffix)
+            logger.info("CMORizing var %s from file %s", var, inpfile)
+            var_info = cmor_table.get_variable(vals['mip'], var)
+            raw_info = {'name': vals['raw'], 'file': inpfile}
+            glob_attrs['mip'] = vals['mip']
+        extract_variable(var_info, raw_info, out_dir, glob_attrs, yr)
```

Comment:
This line is under indented, could be a bug?

Reply:
@bouweandela You are right this should go within the `for` loop. Actually it was not an issue since the loop is done for a single year value. Now it should be correct.